### ISSUE TITLE
L2/R2

### ISSUE THREAD

**Issue by t1ckk** [xerpi/ds4vita#18]
Hi, is it possible to use the ds4 R2/L2 buttons when using moonlight or in dumped games, etc? It says on the latest version 1.1 the plugin adds proper L2/R2 and L3/R3 support but it doesnt seem to be working for me when Im using it. Everything else works just these 4 buttons.

**Comment by t1ckk** [comment]
Ok, I got it to work on adrenaline however on moonlight I'm still unable to use the L2/R2 and L3/R3 buttons.

**Comment by xerpi** [comment]
L2/R2 and L3/R3 are mapped to L2/R2 and L3/R3 and regular PSVita games don't use them.

**Comment by t1ckk** [comment]
Ok, but how about moonlight? I saw people on vitahacks say that L2/R2 should work with moonlight however it doesnt seem to be the case for me.

**Comment by xerpi** [comment]
It's up to moonlight to read L2/R2/R3/L3, I haven't checked its source.

**Comment by t1ckk** [comment]
Oh ok, thanks for the help anyways. Great plugin!

**Comment by Teozbernie** [comment]
Hello xerpi I love your plugin especially for Minecraft of the vita there is just 1 thing that bugs me I can map L3 R3 L2 but for some reason when I press R2 it acts like I have not pressed any buttons my control works fine tested it on my ps4 call of duty I'm also using enso thank you for updating it can you fix this please so I can use R2 would make this absolutely perfect :)

**Comment by xerpi** [comment]
@Teozbernie so L2 works fine but R2 doesn't? And does this only happen with Minecraft or with other games too?

**Comment by Teozbernie** [comment]
@xerpi Yes L2 works great and L3 and R3 but unfortunately R2 is not working at all acts like it's not even there I've only really tried Minecraft because it's the only game where I can remap the buttons to anything I choose I would love it if you could fix this issue

**Comment by Teozbernie** [comment]
@xerpi guess you stopped developing for this app no worries I understand hopefully some day someone with more skills than me can fix the issue :)

**Comment by parkerlreed** [comment]
@Teozbernie That's a bit rude assuming he has stopped development. It's probably just not high priority to fix/reply.

**Comment by Teozbernie** [comment]
@parkerlreed @xerpi I actually managed to fix the issue using the source and compiling it but I've made it so x no longer works now haha don't know how I did this hopefully xerpi can fix it one day my bad if I seemed rude was not my intention at all :)

**Comment by xerpi** [comment]
@Teozbernie would you mind sharing your changes?

**Comment by Teozbernie** [comment]
@xerpi I deleted it but I'll recompile it once I'm back at my PC where should I upload it and can I post the link here? I've worked out a way to use the controller but it's a bit buggy as I use remote play of Minecraft but would really prefer just to play it on the official vita version just let me know where to upload and where to post pal

**Comment by xerpi** [comment]
@Teozbernie just paste a diff to gist or pastebin or wherever you want.

**Comment by Sepalod** [comment]
Can someone explain me how I can compile this plugin. What tools does it need?

**Comment by xerpi** [comment]
@Sepalod you need VITASDK: https://vitasdk.org/

**Comment by Sepalod** [comment]
@xerpi Thxs a lot. I tried it out and worked pretty well. I  switched my l1/r1 with l2/r2. (SCE_CTRL_L1 ->SCE_CTRL_LTRIGGER and same for R1/2). But it's very strange. For Games like NfS Rivals its working pretty good, because now the mapping for R1/2 L1/2 is the same as shown in the game. Same for example for Battlefield 4. But in Games like Diablo 3 or Mortal KOmbat the Mapping is wrong. Here the buttons are mixed up. For example if r1 is shown in the game, i have to press r2 on the ds4 Its very strage. Therefore I created a new issue to be able to configure the mappings for example in a property file? Maybe I can take a look but my knowledge in C is bounded :)

**Comment by 2ea5y4me** [comment]
Hi, got the L2/R2 L3/R3 working with moonlight.

1. Create "vita.conf" file (content is here https://github.com/xyzz/vita-moonlight/blob/vita/mappings/vita.conf )

2. Replace the values for LT, RT, LS, RS:
   btn_tl = 400004
   btn_tr = 400005
   btn_tl2 = 300002
   btn_tr2 = 300004

3. Save the file

4. Copy the file to your vita to: ux0:data/moonlight/mappings/vita.conf

5. Run moonlight and go to "Setings". There set "Enable mapping file" to yes

6. Run streaming
